### ISSUE TITLE
PanelGrid: new layout(-mode) "flex"

### ISSUE THREAD

**Issue by christophs78** [primefaces/primefaces#5159]
PanelGrid has currently the layout-modes "table" and "grid".
Grid is based on the "old" grid-CSS, which has a few corner-cases because it´s based on the CSS-float-mechanism.
Think it would make sense to provide a additional "flex-grid" - layout-mode. (I´ll provide a PR for this within the next month.)

One question: PrimeFlex is currently not provided OOTB with PrimFaces. We may document that this new layout-mode requires PrimeFlex. A other option would be to provide PrimeFlex OOTB starting with PF 7.1.

**Comment by tandraschko** [comment]
Related to #1820

**Comment by melloware** [comment]
@christophs78 Much appreciated for the future PR.  To answer your questions...

1. Assume PrimeTek does not include PrimeFlex out of the box we can add it to the documentation possible even be clever and put a JS Console PrimeFaces.error() if we detect that primeflex.css is not loaded.

2. I would just call the layout mode "flex".

3. I would be amazing if you could fix #1820 while you are adding PrimeFlex!

**Comment by christophs78** [comment]
Probably https://github.com/primefaces/primefaces/issues/1820 was already fixed:
https://github.com/primefaces/primefaces/commit/9f880385fa26287737eb92f79467d232057f40de#diff-d7a99593a8e92d9c69f098585d87dcfa
https://github.com/primefaces/primefaces/issues/2677

**Comment by melloware** [comment]
You are right I just tested and confirmed.  Thanks I closed the other ticket!

**Comment by martin654** [comment]
Hi, can I have a question why **p-nogutter** is used for rows? ( FLEX_ROW_CLASS = "p-grid p-nogutter"; ) What if user wants some gutter? Shouldn't this be optional? Thanks

**Comment by christophs78** [comment]
Do you have a concrete usecase where it would be better without p-nogutter?
When you look at the showcase-page https://www.primefaces.org/showcase/ui/panel/panelGrid.xhtml i think we are fine with p-nogutter.
From a technical point of view it would be easy to remove p-nogutter from FLEX_ROW_CLASS. After introducing layout="flex" we also added the new attribute "contentStyleClass". So it would be easy the add p-nogutter if necessary.

**Comment by martin654** [comment]
Well, in our case we used layout="grid" and switched to layout="flex". In "grid" layout we had configured padding between "cells" of the grid and we wanted to use them as well in "flex" mode. 

In given showcase-page the principle is a bit "wierd" - the theme defines margin that overrides the p-nogutter no-margin. We would have to do the same in our theme. 

I think it would be more straightforward to omit the "p-nogutter" and user could add it using "contentStyleClass" as you mentioned.
